src/components/Header.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodoError } from '../types/TodoError';
+import { USER_ID } from '../api/todos';
+import * as todoService from '../api/todos';
+
+type Props = {
+  todos: Todo[],
+  setErrorMesage: (errorMesage: TodoError) => void,
+  setTodosFromServer: (todos: Todo[]) => void,
+  setNewAddedTodoId: (todoId: number | null) => void,
+  setChangedStatusIds: (ids: number[]) => void,
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setErrorMesage,
+  setTodosFromServer,
+  setNewAddedTodoId,
+  setChangedStatusIds,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const todoTemp = {
+    id: 0,
+    title: todoTitle,
+    completed: false,
+    userId: USER_ID,
+  };
+
+  const addTodo = ({ title, userId, completed }: Todo) => {
+    setIsInputDisabled(true);
+
+    return todoService.createTodo({ title, userId, completed })
+      .then(newTodo => {
+        setTodosFromServer([...todos, newTodo]);
+        setNewAddedTodoId(newTodo.id);
+      })
+      .catch(() => setErrorMesage(TodoError.add))
+      .finally(() => {
+        setTodoTitle('');
+
+        setTimeout(() => {
+          setIsInputDisabled(false);
+          setNewAddedTodoId(null);
+        }, 300);
+      });
+  };
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+
+      if (todoTemp.title.trim() !== '') {
+        addTodo(todoTemp);
+      } else {
+        setErrorMesage(TodoError.emtyTitle);
+      }
+    }
+  };
+
+  const areAllCompleted = () => {
+    return todos.every((todo) => todo.completed);
+  };
+
+  const updateSelectedTodoId = () => {
+    const differentStatusTodos = todos.filter(
+      (todo) => todo.completed === areAllCompleted(),
+    );
+
+    const ids = differentStatusTodos.map((todo) => todo.id);
+
+    setChangedStatusIds(ids);
+  };
+
+  const toggleAllTodos = async () => {
+    const newStatus = !areAllCompleted();
+
+    const updatedTodos = todos.map(todo => {
+      return ({
+        ...todo,
+        completed: newStatus,
+      });
+    });
+
+    try {
+      await Promise.all(
+        updatedTodos.map(updateTodo => todoService.updateTodo(updateTodo)),
+      );
+      setTodosFromServer(updatedTodos);
+      updateSelectedTodoId();
+    } catch {
+      setErrorMesage(TodoError.update);
+      setTodosFromServer(todos);
+    } finally {
+      setTimeout(() => setChangedStatusIds([]), 300);
```

Comment:
don't use setTimeout
```suggestion
      setChangedStatusIds([]);
```